lib/commands/performance.js

```diff
@@ -0,0 +1,339 @@
+import { errors } from 'appium-base-driver';
+import { sleep } from 'asyncbox';
+import _ from 'lodash';
+
+let commands = {}, helpers = {}, extensions = {};
+
+commands.postDismissAlert = async function () {
+  throw new errors.NotYetImplementedError();
+};
+
+const SUPPORTED_PERFORMANCE_DATA_TYPES = {
+  cpuinfo: 'the amount of cpu by user and kernel process - cpu information for applications on real devices and simulators',
+  memoryinfo: 'the amount of memory used by the process - memory information for applications on real devices and simulators',
+  batteryinfo: 'the remaining battery power - battery power information for applications on real devices and simulators',
+  networkinfo: 'the network statistics - network rx/tx information for applications on real devices and simulators'
+};
+
+//
+// returns the information type of the system state which is supported to read as like cpu, memory, network traffic, and battery.
+// output - array like below
+//[cpuinfo, batteryinfo, networkinfo, memoryinfo] 
+//
+commands.getPerformanceDataTypes = function () {
+  return _.keys(SUPPORTED_PERFORMANCE_DATA_TYPES);
+};
+
+// returns the information type of the system state which is supported to read as like cpu, memory, network traffic, and battery.
+//input - (packageName) the package name of the application 
+//        (dataType) the type of system state which wants to read. It should be one of the keys of the SUPPORTED_PERFORMANCE_DATA_TYPES
+//        (dataReadTimeout) the number of attempts to read
+// output - table of the performance data, The first line of the table represents the type of data. The remaining lines represent the values of the data.
+//
+// in case of battery info : [[power], [23]]
+// in case of memory info : [[totalPrivateDirty, nativePrivateDirty, dalvikPrivateDirty, eglPrivateDirty, glPrivateDirty, totalPss, nativePss, dalvikPss, eglPss, glPss, nativeHeapAllocatedSize, nativeHeapSize], [18360, 8296, 6132, null, null, 42588, 8406, 7024, null, null, 26519, 10344]]
```

Comment:
If I'm reading this correctly, this returns a "table", which is an array of arrays, in which the first array is keys and the second array is the corresponding values? So for memory info it gives

```
[
  [totalPrivateDirty, nativePrivateDirty, dalvikPrivateDirty, eglPrivateDirty, glPrivateDirty, totalPss, nativePss, dalvikPss, eglPss, glPss, nativeHeapAllocatedSize, nativeHeapSize], 
  [18360, 8296, 6132, null, null, 42588, 8406, 7024, null, null, 26519, 10344]
]
```

Is there no way to make it as associative array? Since this is an entirely new API, why not have it as simple to use as possible. So the above would be

```
{
  totalPrivateDirty: 18360, 
  nativePrivateDirty: 8296, 
  dalvikPrivateDirty: 6132, 
  eglPrivateDirty: null, 
  glPrivateDirty: null, 
  totalPss: 42588, 
  nativePss: 8406, 
  dalvikPss: 7024, 
  eglPss: null, 
  glPss: null, 
  nativeHeapAllocatedSize: 26519, 
  nativeHeapSize: 10344
}
```